Feature/proj 69

Lelik7777 opened the pull request:
# Feature: SignIn and AuthCard Enhancements

## Overview

This pull request introduces several enhancements and new features related to the SignIn and AuthCard components. It includes the creation of new folders with templates, updates to component logic, styles, and documentation improvements. Additionally, it involves updating dependencies to ensure compatibility and improved functionality.

## Changes

### SignIn Component

- **Folder Structure**: Added a new `signIn` folder with relevant templates to organize the SignIn component files.
- **Layout and Styles**: Introduced a new layout and updated styles for the SignIn component.
  - Updated `_variables.scss` to include new style variables.
  - Updated `signIn.module.scss` for improved styling.
- **Component Updates**:
  - Updated `signIn.tsx` to enhance functionality and remove unused imports.
  - Added `ErrorMessage` and `FormButtons` components for better error handling and form actions.
  - Integrated `useSignIn` hook for managing form state and logic.
- **Dependency Updates**:
  - Installed `react-hook-form` for form management.
  - Updated `@photo-fiesta/ui-lib` to version 0.0.6.
  - Updated `@hookform/resolvers` to version 3.9.0.
  - Installed `zod` version 3.23.8 for schema validation.

### AuthCard Component

- **Folder Structure**: Added a new `authCard` folder with relevant templates to organize the AuthCard component files.
- **Logic Relocation**: Moved certain logic from the SignIn component to the AuthCard component for better modularity and reusability.
- **Component Updates**:
  - Updated `authCard.tsx` to include JSDoc comments for better documentation and understanding.
  - Updated `authCard.module.scss` for improved styling.

### Refactoring

- **Index File**: Updated `index.tsx` where the AuthCard component is rendered to reflect the changes and improvements.

## Documentation

- Added JSDoc comments to both the `AuthCard` and `SignIn` components, providing detailed descriptions and usage examples.

## Testing

- [ ] Ensure all components render correctly with the updated styles and logic.
- [ ] Verify that form submission and validation work as expected with the new `useSignIn` hook and `zod` validation.
- [ ] Test the integration of the updated dependencies to ensure compatibility.

## Screenshot
![Screenshot_1](https://github.com/user-attachments/assets/357efa39-45c5-4064-9389-013631f759a4)

Lelik7777 committed:
feature(signIn): add signIn folder with templates

Lelik7777 committed:
feature(signIn):add signIn layout and styles

- Install react-hook-form
- Update _variables.scss

Lelik7777 committed:
feature(signIn):update signIn component

- add useSignIn hook
- update signIn.module.scss
- update signIn.tsx
- update @photo-fiesta/ui-lib to 0.0.6
- update @hookform/resolvers to 3.9.0
- install zod to 3.23.8

Lelik7777 committed:
feature(authCard): add authCard folder with templates

Lelik7777 committed:
feature: relocate logic from signIn to authCard

- Update signIn.tsx: remove unused imports, add ErrorMessage, FormButtons
- Update signIn.module.scss
- Update authCard.tsx
- Update authCard.module.scss

Lelik7777 committed:
refactor: update index.tsx where auth card is rendered

- Add js doc comments to authCard component and signIn component

Lelik7777 committed:
Merge branch 'dev' into feature/proj-69

EvgeniyaYakunina commented on src/features/auth/ui/authCard/authCard.module.scss:
```diff
@@ -0,0 +1,32 @@
+  flex-direction: column;
+  gap: 6px;
+  align-items: center;
+}
```
Добавить адаптив для мобильных устройств

Lelik7777 replied:
добавим через отдельную таску после мержа proj-82

EvgeniyaYakunina commented on src/features/auth/ui/signIn/signIn.tsx:
```diff
@@ -0,0 +1,108 @@
+    container: styles.container,
+    form: styles.form,
+  } as const
+  const isButtonDisabled = !isValid || !isDirty
```
Проверить необходимость этой константы, так как по макету кнопка не дизейблится

Lelik7777 replied:
исправил

EvgeniyaYakunina commented on src/features/auth/ui/signIn/signIn.tsx:
```diff
@@ -0,0 +1,108 @@
+
+  return (
+    <AuthCard
+      footerLinkHref={'/sign-up'}
```
нужна todo, так как линки нужно будет проверить после того как определимся с нэймингом путей.

Lelik7777 replied:
исправил

EvgeniyaYakunina commented on src/features/auth/ui/signIn/signIn.tsx:
```diff
@@ -0,0 +1,108 @@
+          />
+          {(errors.email || errors.password) && <ErrorMessage />}
+        </div>
+        <FormButtons isButtonDisabled={isButtonDisabled} />
```
А откуда у нас такая компонента FormButtons?  Везде используется просто кнопка сабмит....

Lelik7777 replied:
это блок с кнопками, вынес его отдельно. В нем присутствует <Button variant={'primary'}>Sign in</Button>, которая и реализует submit

EvgeniyaYakunina commented on src/features/auth/ui/signIn/useSignIn.ts:
```diff
@@ -0,0 +1,44 @@
+    .string()
+    .min(6, 'Password must be at least 8 characters')
+    .max(20, 'Password must be at most 32 characters')
+    .regex(/^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*()_+\-=\\[\]{};':"\\|,.<>/?]).*$/),
```
создала для regex константу в папке shared/config

Lelik7777 replied:
добавим через отдельную таску после мержа proj-82

Lelik7777 committed:
refactor(signIn):update signIn component

- Remove    isDirty, isValid
- Remove disabled for button signIn
- Change path for import _mixins:  @use '@/shared/styles/mixins' as *
- Update js commit message in mixins

Lelik7777 committed:
Merge branch 'dev' into feature/proj-69

Lelik7777 committed:
chore:add commits in useSignIn and signIn.module.scss

Lelik7777 committed:
Merge branch 'feature/proj-69' of https://github.com/NoName-ForTeam/photo-fiesta into feature/proj-69

Lelik7777 committed:
refactor:change import paths in auth card and sign up

Lelik7777 committed:
fix:fix eslint errors in useSignUpForm
